test/hotspot/jtreg/compiler/loopopts/superword/MinMaxRed_Long.java

```diff
@@ -0,0 +1,155 @@
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+ * @test
+ * @bug 8307513
+ * @summary [SuperWord] MaxReduction and MinReduction should vectorize for long
+ * @library /test/lib /
+ * @run driver compiler.loopopts.superword.MinMaxRed_Long
+ */
+
+package compiler.loopopts.superword;
+
+import compiler.lib.ir_framework.*;
+import jdk.test.lib.Utils;
+
+import java.util.Arrays;
+import java.util.Random;
+import java.util.stream.LongStream;
+
+public class MinMaxRed_Long {
+
+    private static final Random random = Utils.getRandomInstance();
+
+    public static void main(String[] args) throws Exception {
+        TestFramework framework = new TestFramework();
+        framework.addFlags("-XX:+IgnoreUnrecognizedVMOptions",
+                           "-XX:LoopUnrollLimit=250",
+                           "-XX:CompileThresholdScaling=0.1");
+        framework.start();
+    }
+
+    @Run(test = {"maxReductionImplement"},
+         mode = RunMode.STANDALONE)
+    public void runMaxTest() {
+        runMaxTest(50);
+        runMaxTest(80);
+        runMaxTest(100);
+    }
+
+    private static void runMaxTest(int probability) {
+        long[] longs = new long[1024];
+        ReductionInit(longs, probability);
+        long res = 0;
+        for (int j = 0; j < 2000; j++) {
+            res = maxReductionImplement(longs, res);
+        }
+        if (res == 11 * Arrays.stream(longs).max().getAsLong()) {
+            System.out.println("Success");
+        } else {
+            throw new AssertionError("Failed");
+        }
+    }
+
+    @Run(test = {"minReductionImplement"},
+         mode = RunMode.STANDALONE)
+    public void runMinTest() {
+        runMinTest(50);
+        runMinTest(80);
+        runMinTest(100);
+    }
+
+    private static void runMinTest(int probability) {
+        long[] longs = new long[1024];
+        ReductionInit(longs, probability);
+        // Negating the values generated for controlling max branching
+        // allows same logic to be used for min tests.
+        longs = negate(longs);
+        long res = 0;
+        for (int j = 0; j < 2000; j++) {
+            res = minReductionImplement(longs, res);
+        }
+        if (res == 11 * Arrays.stream(longs).min().getAsLong()) {
+            System.out.println("Success");
+        } else {
+            throw new AssertionError("Failed");
+        }
+    }
+
+    static long[] negate(long[] nums) {
+        return LongStream.of(nums).map(l -> -l).toArray();
+    }
+
+    public static void ReductionInit(long[] longs, int probability) {
```

Comment:
```suggestion
    public static void reductionInit(long[] longs, int probability) {
```
This is a method name, not a class - so I think it should start lower-case, right?

Reply:
And the method might as well allocate the array too. But up to you.